contracts/src/Tracker.sol

```diff
@@ -16,103 +17,109 @@ struct Challenge {
     address public underlyingToken;
+    uint256 public challengeCounter;
 
-    mapping(address arxAddress => Challenge) public challenges;
-    mapping(address userAddress => address[]) public userChallenges;
+    mapping(uint256 challengeId => Challenge) public challenges;
+    mapping(address userAddress => uint256[]) public userChallenges;
     mapping(address userAddress => uint256) public balances;
-    mapping(address arxAddress => mapping(address userAddress => uint256[])) public checkIns;
-    mapping(address arxAddress => mapping(address userAddress => bool)) public hasJoined;
-    mapping(address arxAddress => address[]) public participants;
-    mapping(address arxAddress => address[]) public succeedParticipants;
+    mapping(uint256 challengeId => mapping(address userAddress => uint256[])) public checkIns;
+    mapping(uint256 challengeId => mapping(address userAddress => bool)) public hasJoined;
+    mapping(uint256 challengeId => address[]) public participants;
+    mapping(uint256 challengeId => address[]) public succeedParticipants;
+    mapping(bytes32 digest => bool) public digestUsed;
 
-    event CheckIn(address indexed userAddress, address indexed arxAddress, uint256 timestamp);
+    event CheckIn(address indexed userAddress, uint256 indexed challengeId, uint256 timestamp);
     event Register(
-        address indexed arxAddress,
+        uint256 indexed challengeId,
+        address verifierAddress,
         string description,
         uint256 startTimestamp,
         uint256 endTimestamp,
         uint256 minimumCheckIns
     );
-    event Join(address indexed userAddress, address indexed arxAddress);
-    event Settle(address indexed arxAddress);
+    event Join(address indexed userAddress, uint256 indexed challengeId);
+    event Settle(uint256 indexed challengeId);
 
-    constructor(address _underlyingToken) {
+    constructor(address _underlyingToken, string memory name, string memory version) EIP712(name, version) {
         //FIXME: This should be moved to register
         underlyingToken = _underlyingToken;
     }
 
     //FIXME: This should be created with token specify
     // register a new habit challenge
     function register(
-        address arxAddress,
+        address verifierAddress,
         string memory description,
         uint256 minimunCheckIns,
         uint256 startTimestamp,
         uint256 endTimestamp,
         address donateDestination,
         uint256 stake
     ) public {
-        require(challenges[arxAddress].startTimestamp == 0, "Challenge already exists");
         //require(endTimestamp > startTimestamp, "End timestamp must be greater than start timestamp");
-        challenges[arxAddress] =
-            Challenge(minimunCheckIns, startTimestamp, endTimestamp, donateDestination, stake, 0, false);
-        emit Register(arxAddress, description, startTimestamp, endTimestamp, minimunCheckIns);
+        challenges[challengeCounter] = Challenge(
```

Comment:
you can just use 
```
challenges[challengeCounter++];
```
So it returns the old counter and += 1;
(then you can remove challenge ++)